modules/plotting/processors/dataframeexporter.cpp

```diff
@@ -94,9 +94,13 @@ void DataFrameExporter::exportNow() {
         LogWarn("File already exists: " << exportFile_);
         return;
     }
-    if (exportFile_.getSelectedExtension() == exportFile_.getNameFilters()[0]) {
+    // convert the fileEnding to all lower case, for string comparison a few lines down.
+    std::string fileEnding = filesystem::getFileExtension(exportFile_.get());
+    std::transform(fileEnding.begin(), fileEnding.end(), fileEnding.begin(), ::tolower);
+
+    if (fileEnding == "csv") {
         exportAsCSV(separateVectorTypesIntoColumns_);
-    } else if (exportFile_.getSelectedExtension() == exportFile_.getNameFilters()[1]) {
+    } else if (fileEnding == "xml") {
```

Comment:
there is a slight semantic difference with here, exportFile_.getSelectedExtension() can return different readers for the same extension, the given solution will not handle that. but since there is only csv and xml that should not matter here

Reply:
Martin is working on it...